apps/dashboard/src/components/variable/variable-list.tsx

```diff
@@ -151,3 +142,53 @@ export const VariableList = React.forwardRef<VariableListRef, VariablesListProps
     );
   }
 );
+
+const VariableListItem = ({
+  option,
+  index,
+  selectedValue,
+  hoveredOptionIndex,
+  setHoveredOptionIndex,
+  onSelect,
+}: {
+  option: VariablesListProps['options'][number];
+  index: number;
+  selectedValue?: string;
+  hoveredOptionIndex: number;
+  setHoveredOptionIndex: (index: number) => void;
+  onSelect: (value: string) => void;
+}) => {
+  const hasPreview = !!option.preview;
+  return (
+    <Tooltip open={hoveredOptionIndex === index && hasPreview} key={option.value}>
+      <TooltipTrigger asChild>
+        <li
+          className={cn(
+            'text-paragraph-xs font-code text-foreground-950 flex cursor-pointer items-center gap-1 rounded-sm p-1 hover:bg-neutral-100',
+            hoveredOptionIndex === index ? 'bg-neutral-100' : ''
+          )}
+          value={option.value}
+          onClick={(e) => {
+            e.stopPropagation();
+            e.preventDefault();
+
+            onSelect(option.value ?? '');
+          }}
+          onMouseEnter={() => setHoveredOptionIndex(index)}
+          onMouseLeave={() => setHoveredOptionIndex(-1)}
+        >
+          <div className="flex size-3 items-center justify-center">
+            {hasPreview ? <DigestVariableIcon className="size-full" /> : <Code2 className="text-feature size-full" />}
```

Comment:
I did some minor change in my PR regarding this:

https://github.com/novuhq/novu/pull/8125/files#diff-67a70e79211ae258ff99da6ad6de04d6a4795995c9321a1c92cc7e1831963a47R134

I just extracted the variable icons to separate component cause I also needed to add new icon - maybe you could cherry-pick that component so we dont have a conflict

Reply:
Thanks, I will do that 🙌🏼